modules/generative-octoai/module.go

```diff
@@ -0,0 +1,97 @@
+//                           _       _
+// __      _____  __ ___   ___  __ _| |_ ___
+// \ \ /\ / / _ \/ _` \ \ / / |/ _` | __/ _ \
+//  \ V  V /  __/ (_| |\ V /| | (_| | ||  __/
+//   \_/\_/ \___|\__,_| \_/ |_|\__,_|\__\___|
+//
+//  Copyright © 2016 - 2024 Weaviate B.V. All rights reserved.
+//
+//  CONTACT: hello@weaviate.io
+//
+
+package modgenerativeoctoai
+
+import (
+	"context"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/weaviate/weaviate/entities/modulecapabilities"
+	"github.com/weaviate/weaviate/entities/moduletools"
+	"github.com/weaviate/weaviate/modules/generative-octoai/clients"
+	additionalprovider "github.com/weaviate/weaviate/usecases/modulecomponents/additional"
+	generativemodels "github.com/weaviate/weaviate/usecases/modulecomponents/additional/models"
+)
+
+const Name = "generative-octoai"
+
+func New() *GenerativeOctoAIModule {
+	return &GenerativeOctoAIModule{}
+}
+
+type GenerativeOctoAIModule struct {
+	generative                   generativeClient
+	additionalPropertiesProvider modulecapabilities.AdditionalProperties
+}
+
+type generativeClient interface {
+	GenerateSingleResult(ctx context.Context, textProperties map[string]string, prompt string, cfg moduletools.ClassConfig) (*generativemodels.GenerateResponse, error)
+	GenerateAllResults(ctx context.Context, textProperties []map[string]string, task string, cfg moduletools.ClassConfig) (*generativemodels.GenerateResponse, error)
+	Generate(ctx context.Context, cfg moduletools.ClassConfig, prompt string) (*generativemodels.GenerateResponse, error)
+	MetaInfo() (map[string]interface{}, error)
+}
+
+func (m *GenerativeOctoAIModule) Name() string {
+	return Name
+}
+
+func (m *GenerativeOctoAIModule) Type() modulecapabilities.ModuleType {
+	return modulecapabilities.Text2TextGenerative
+}
+
+func (m *GenerativeOctoAIModule) Init(ctx context.Context,
+	params moduletools.ModuleInitParams,
+) error {
+	if err := m.initAdditional(ctx, params.GetConfig().ModuleHttpClientTimeout, params.GetLogger()); err != nil {
+		return errors.Wrap(err, "init q/a")
+	}
+
+	return nil
+}
+
+func (m *GenerativeOctoAIModule) initAdditional(ctx context.Context, timeout time.Duration,
+	logger logrus.FieldLogger,
+) error {
+	apiKey := os.Getenv("OCTOAI_API_KEY")
```

Comment:
Let's use `OCTOAI_APIKEY` instead of `OCTOAI_API_KEY`

Reply:
ack